core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -112,7 +114,10 @@ class KafkaRequestHandler(
       val req = requestChannel.receiveRequest(300)
       val endTime = time.nanoseconds
       val idleTime = endTime - startSelectTime
-      aggregateIdleMeter.mark(idleTime / totalHandlerThreads.get)
+      // Per-pool idle ratio uses the pool's own thread count as denominator
+      perPoolIdleMeter.mark(idleTime / poolHandlerThreads.get)
+      // Aggregate idle ratio uses the total threads across all pools as denominator
+      aggregateIdleMeter.mark(idleTime / aggregateThreads.get)
```

Comment:
It is an inconsistent design that the `aggregateIdleMeter` is defined in the constructor while the `aggregateThreads` is defined in an object. Why is that? Why not pass the aggregateThreads through the constructor?